### PR TITLE
Modernise cheat sheet

### PR DESCRIPTION
**Pull request by jrha**
Bring example Pan snippets up to date with current coding practices.

### CODE REVIEW

File: _posts/2014-02-26-cheat-sheet.md [hunk 1]
```diff
@@ -42,7 +28,7 @@ Pin version:
 Pin Architecture:
 
 ```sh
-"/software/packages/{config-templates-metaconfig.noarch}" ?= nlist();
+"/software/packages/{config-templates-metaconfig.noarch}" ?= dict();
```
Comment (stdweird) [inline _posts/2014-02-26-cheat-sheet.md]: is it possible to use a packge that is actually used? or make avery clear dummy name like `{acme-tools}`

File: _posts/2014-02-26-cheat-sheet.md [hunk 2]
```diff
@@ -76,14 +62,12 @@ Ensure a directory exists (and has correct permissions)
...
 ```sh
-'/software/components/dirperm/paths' = append(
-    nlist(
-        'path', "/opt/temple",
-        'owner', "syrinx:syrinx",
-        'perm', '0755',
-        'type', 'd',
-    )
-);
+'/software/components/dirperm/paths' = append(SELF, dict(
```
Comment (stdweird) [inline _posts/2014-02-26-cheat-sheet.md]: stricly speaking, the `SELF` is not needed

Comment (jrha) [inline _posts/2014-02-26-cheat-sheet.md]: It's not needed, but I consider it best practice to include it, users who are used to `push()` sometimes try to add a second item and then wonder why existing items on the left-hand-side disappear.

**Comment by jrha** [comment]
Tests are failing on a different page

File: _posts/2014-02-26-cheat-sheet.md [hunk 3]
```diff
@@ -12,43 +13,28 @@ If you have any other examples, please add them!
 
 Add default (latest) version of a package
 =========================================
-rpmt based spma (deprecated)
-----------------------------
-
 ```sh
 '/software/packages'=pkg_repl('ruby');
```
Comment (stdweird) [inline _posts/2014-02-26-cheat-sheet.md]: panlint failure, needs spaces

Comment (jrha) [inline _posts/2014-02-26-cheat-sheet.md]: I should figure out a way to get panlint to lint code blocks in markdown documents...

**Pull request merged**